src/features/defi/providers/idle/components/IdleManager/Withdraw/components/Confirm.tsx

```diff
@@ -132,11 +142,18 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
       )
         return
       dispatch({ type: IdleWithdrawActionType.SET_LOADING, payload: true })
-      const idleOpportunity = await idleInvestor.findByOpportunityId(opportunityData?.assetId)
       if (!idleOpportunity) throw new Error('No opportunity')
+
+      const idleAssetWithdrawAmount = bnOrZero(state.withdraw.cryptoAmount).div(
+        idleOpportunity.positionAsset.underlyingPerPosition,
+      )
+      // TODO: This is fine for now, but going forward we will need to:
+      // 1. Use base unit amounts everywhere, we shouldn't need to `.times(asset.precision)
+      // 2. Have a notion of a "display amount" and "underlying amount"
+      // These two notions should be decoupled and we should *not* have to calc back and forth from the wrapping and underlying assets
       const tx = await idleOpportunity.prepareWithdrawal({
         address: userAddress,
-        amount: bnOrZero(state.withdraw.cryptoAmount).times(`1e+${asset.precision}`).integerValue(),
+        amount: idleAssetWithdrawAmount.times(`1e+${asset.precision}`).integerValue(),
```

Comment:
Isn't this just `toBaseUnit(idleAssetWithdrawAmount, asset.precision)`?

Reply:
Ab-solutely, and safer than `integerValue()` as well 🙏🏽 confirmed we're getting the same value, wrapped it into a BigNumber to honor `investor-idle` types

<img width="493" alt="image" src="https://user-images.githubusercontent.com/17035424/203989858-e6b7f9b8-861a-46b0-9c00-a4b31abc7f24.png">
<img width="586" alt="image" src="https://user-images.githubusercontent.com/17035424/203991149-f7538ef6-a849-4145-9f66-8c126ad05cc3.png">